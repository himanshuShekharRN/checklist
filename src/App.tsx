import React from 'react';
import 'react-native-gesture-handler';

import {Navigation} from './navigation';
import {Provider} from 'react-redux';
import {persistor, store} from './store';
import {PersistGate} from 'redux-persist/integration/react';

export const App: React.FC = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Navigation />
      </PersistGate>
    </Provider>
  );
};
