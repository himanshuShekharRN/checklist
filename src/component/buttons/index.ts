import {FloatingAddButton} from './floatingAddButton';
import {RoundedButton} from './roundedButton';

export {FloatingAddButton, RoundedButton};
