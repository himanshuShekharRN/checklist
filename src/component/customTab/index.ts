import {CustomTab} from './CustomTab';
import {FirstRoute} from './firstRoute';
import {SecondRoute} from './secondRoute';
import {ThirdRoute} from './thirdRoute';
import {FourthRoute} from './fourthRoute';
import {FifthRoute} from './fifthRoute';

export {
  CustomTab,
  FirstRoute,
  SecondRoute,
  ThirdRoute,
  FourthRoute,
  FifthRoute,
};
