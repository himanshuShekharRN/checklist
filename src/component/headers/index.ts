import {HeadersWithButton} from './headersWithButton/HeadersWithButton';

export {HeadersWithButton};
