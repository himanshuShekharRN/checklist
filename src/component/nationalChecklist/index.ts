import {NationalChecklist} from './NationalChecklist';

export {NationalChecklist};
