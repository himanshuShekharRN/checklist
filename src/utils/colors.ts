export const COLOR_BLACK = '#000000';
export const COLOR_WHITE = '#FFFFFF';
export const COLOR_NAVY_300 = '#ECEEF0';
