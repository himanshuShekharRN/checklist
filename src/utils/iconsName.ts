export const OUTLINE_EXCLAMATION = 'error-outline';
export const WORKSPACE_PREMIUM = 'workspace-premium';
export const FILE_FILLED = 'description';
export const DONE = 'check-circle';
export const SKIPPED_OPTIONAL = 'do-disturb-on';
export const PENDING = 'info';
export const UNCHECK = 'panorama-fish-eye';
export const DELETE = 'delete';
export const RIGHT_ANGLE_BRACKET = 'keyboard-arrow-right';
