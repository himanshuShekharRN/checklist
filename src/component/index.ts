import {SwipableCards} from './swipableCards/SwipableCards';
import {SwipableActionButton} from './swipableActionButton/SwipableActionButton';
import {Card} from './cards';
import {MyCheckList} from './myChecklist';
import {Space} from './space';
import {PreDepartureList} from './preDepartureList';
import {Title} from './title';
import {HeadersWithButton} from './headers';
import {CustomSVG} from './customSVG';
import {CustomIcon} from './customIcon';
import {FloatingAddButton, RoundedButton} from './buttons';
import {ListCard} from './listCard';
import {LinearProgressBar} from './progressBar';
import {CustomTab} from './customTab';
import {DepartureDocCard} from './departureDocCard';
import {NationalChecklist} from './nationalChecklist';

export {
  FloatingAddButton,
  Card,
  MyCheckList,
  Space,
  PreDepartureList,
  Title,
  HeadersWithButton,
  CustomSVG,
  CustomIcon,
  RoundedButton,
  ListCard,
  LinearProgressBar,
  CustomTab,
  DepartureDocCard,
  NationalChecklist,
  SwipableActionButton,
  SwipableCards,
};
