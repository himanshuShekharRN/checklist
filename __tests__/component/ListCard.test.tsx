import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {ListCard} from '../../src/component';
import {RIGHT_ANGLE_BRACKET} from '../../src/utils/iconsName';
import {COLOR_BLACK} from '../../src/utils/colors';

describe('ListCard', () => {
  const mockFunction = jest.fn();
  const textStyle = {
    color: COLOR_BLACK,
  };
  const props = {
    iconName: RIGHT_ANGLE_BRACKET,
    editable: false,
    readonly: true,
    textStyle: textStyle,
    listTitle: 'Test List Title',
    onChangeTextHandler: mockFunction,
  };

  const tree = render(<ListCard {...props} />);

  test('renders correctly', () => {
    expect(tree.toJSON()).toBeTruthy();
  });

  test('renders title correctly', () => {
    render(<ListCard {...props} />);

    const titleContent = screen.getByText(props.listTitle);
    expect(titleContent).toBeTruthy();
  });

  test('renders TextInput correctly', () => {
    const _props = {
      iconName: RIGHT_ANGLE_BRACKET,
      editable: true,
      readonly: false,
      textStyle: textStyle,
      listTitle: 'Test List Title',
      onChangeTextHandler: mockFunction,
    };
    render(<ListCard {..._props} />);

    const textInput = screen.getByTestId('listCardTextInput');
    expect(textInput).toBeTruthy();
  });

  test('applies custom textStyle', () => {
    render(<ListCard {...props} />);

    const text = screen.getByTestId('listCardText');
    expect(text.props.style).toContainEqual(textStyle);
  });

  test('snapshot created successfully', () => {
    render(<ListCard {...props} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
