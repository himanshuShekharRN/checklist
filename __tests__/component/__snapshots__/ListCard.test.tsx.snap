// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCard snapshot created successfully 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderBottomColor": "#CCCDCF",
      "borderBottomWidth": 1,
      "borderTopColor": "#CCCDCF",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 16,
      "paddingVertical": 22,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#005AA5",
        "borderRadius": 15,
        "height": 30,
        "justifyContent": "center",
        "marginRight": 16,
        "width": 30,
      }
    }
    testID="customIcon"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 12,
          },
          undefined,
          {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      îŒ•
    </Text>
  </View>
  <Text
    style={
      [
        {
          "color": "#000000",
          "flex": 1,
          "fontFamily": "Roboto-Regular",
          "fontSize": 16,
        },
        {
          "color": "#000000",
        },
      ]
    }
    testID="listCardText"
  >
    Test List Title
  </Text>
</View>
`;
